subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/Interpreter.kt

```diff
@@ -477,5 +509,19 @@ fun maybeUnwrapInvocationTargetException(e: Throwable) =
     else e
 
 
+// TODO:partial-evaluator restore contextClassLoader behaviour
+private
+inline fun withContextClassLoader(classLoader: ClassLoader, block: () -> Unit) {
+    val currentThread = Thread.currentThread()
+    val previous = currentThread.contextClassLoader
+    try {
+        currentThread.contextClassLoader = classLoader
+        block()
+    } finally {
+        currentThread.contextClassLoader = previous
+    }
+}
```

Comment:
Can you explain what's going on here a bit more? Are you just able to swap out the class-loader on a thread like this?